### PR TITLE
[KYUUBI #2323] Separate events to a submodule - kyuubi-event

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

Separate events to a submodule - kyuubi-event

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-hive-sql-engine/pom.xml [hunk 1]
```diff
@@ -46,6 +46,12 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
```
Comment (yaooqinn) [inline externals/kyuubi-hive-sql-engine/pom.xml:49]: shade

**Comment by yaooqinn** [comment]
lets also add a log4j file in test resource

File: externals/kyuubi-hive-sql-engine/pom.xml (hunk 1 above)
Comment (zhaomin1423) [reply externals/kyuubi-hive-sql-engine/pom.xml:49]: done

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    password: String,
+    ipAddress: String,
+    conf: Map[String, String],
+    sessionManager: KyuubiSessionManager)
+  extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
+
+  protected val sessionEvent = KyuubiSessionEvent(this)
+  EventBus.post(sessionEvent)
+
+  def getSessionEvent: Option[KyuubiEvent] = {
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala:35]: shall we just return `Option[KyuubiSessionEvent]` ?

File: pom.xml [hunk 3]
```diff
@@ -84,6 +84,7 @@
         <module>kyuubi-metrics</module>
         <module>kyuubi-server</module>
         <module>kyuubi-zookeeper</module>
+        <module>kyuubi-events</module>
```
Comment (ulysses-you) [inline pom.xml:87]: nit: name ordering

File: kyuubi-events/pom.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>kyuubi-events_2.12</artifactId>
+    <name>Kyuubi Project Events</name>
+    <packaging>jar</packaging>
+    <url>https://kyuubi.apache.org/</url>
+
+    <dependencies>
+        <dependency>
```
Comment (yaooqinn) [inline kyuubi-events/pom.xml:35]: we can remove json related deps from common and add them here

Comment (zhaomin1423) [reply kyuubi-events/pom.xml:35]: json also is used in other class, for example, org.apache.kyuubi.service.authentication.KyuubiInternalAccessIdentifier in common module.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala (hunk 2 above)
Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala:35]: do we define only an empty getSession method in KyuubiSession class?

File: kyuubi-events/pom.xml (hunk 4 above)
Comment (yaooqinn) [reply kyuubi-events/pom.xml:35]: OK, let's leave it then. the original purpose of separating these two is to remove json libs from common.
Let's leave it in this PR, and see if we can remove it from KyuubiInternalAccessIdentifier

File: pom.xml (hunk 3 above)
Comment (zhaomin1423) [reply pom.xml]: done

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala (hunk 2 above)
Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSession.scala]: thanks, done

**Comment by zhaomin1423** [comment]
> lets also add a log4j file in test resource

done

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**